pytorch_forecasting/metrics/tests/test_all_metrics.py

```diff
@@ -0,0 +1,449 @@
+
+        The test does not attempt to replicate the internal implementation of
+        quantile prediction generation inside the metric class and equate it with the
+        provided `quantile_pred` since the generation of quantiles in case of
+        distribution losses is a stochastic process, and the quantile prediction
+        generated on the distribution data may never match with our `quantile_pred`
+        tensor.
+
+        Parameters
+        ----------
+        metric: Metric
+            The metric instance that overrides `to_quantiles`.
+        y_pred: torch.Tensor
+            The predicted values tensor.
+        quantiles: list
+            A list of quantiles to be tested.
+        """
+
+        # the reason we have a separate case of mqf2 is because
+        # it produces outputs from `to_quantiles` only on the
+        # input `prediction_length` and not on the entire sequence
+        # length of the `y_pred` tensor.
+
+        assert quantile_pred.shape == (
+            y_pred.shape[0],
+            y_pred.shape[1],
+            len(quantiles),
+        ), f"`to_quantiles()`, fails for metric {metric}."
+
+    def _test_to_quantile_mqf2(self, metric, y_pred, quantile_pred, quantiles):
+        """Test the usage of `to_quantiles` method for MQF2 distribution loss metric.
+
+        The test does not attempt to replicate the internal implementation of
+        quantile prediction generation inside the metric class and equate it with the
+        provided `quantile_pred` since the generation of quantiles in case of
+        distribution losses is a stochastic process, and the quantile prediction
+        generated on the distribution data may never match with our `quantile_pred`
+        tensor.
+
+        Parameters
+        ----------
+        metric: Metric
+            The metric instance that overrides `to_quantiles`.
+        y_pred: torch.Tensor
+            The predicted values tensor.
+        quantiles: list
+            A list of quantiles to be tested.
+        """
+
+        # the reason we have a separate case of mqf2 is because
+        # it produces outputs from `to_quantiles` only on the
+        # input `prediction_length` and not on the entire sequence
+        # length of the `y_pred` tensor.
+
+        assert quantile_pred.shape == (
+            y_pred.shape[0],
+            metric.prediction_length,
+            len(quantiles),
+        ), f"`to_quantiles()`, fails for metric {metric}."
+
+    def _test_integration_metrics(self, metric, y_pred, y, object_pkg):
+        metric.reset()
+        metric.update(y_pred, y)
+        res = metric.compute()
+        assert isinstance(res, torch.Tensor)
+        assert torch.isfinite(res).all(), "Non-finite values in metric result."
+
+        point_pred = metric.to_prediction(y_pred)
+        assert isinstance(point_pred, torch.Tensor), "Prediction should be a tensor."
+        if object_pkg.get_class_tag("info:metric_name") == "MQF2DistributionLoss":
```

Comment:
this functions has imo too many case distinctions and individual stepouts. Can we avoid the individual stepouts?

Some cases looks like a non-conformance, actually.